### PR TITLE
Change ClientActor to prioritize block production during high load

### PR DESCRIPTION
**Pull request by pmnoxx**
``ClientActor`` will try to check if it's time to do block production every time it receives messages from inbox in ``handle`` method. This will ensure that block production still happens during high load.

Context:
There is a bug in Actix library. While there are messages in mailbox, Actix
will prioritize processing messages until mailbox is empty. Execution of any other task
scheduled with run_later will be delayed.

### CODE REVIEW

File: chain/client/src/client_actor.rs [hunk 1]
```diff
@@ -175,6 +179,11 @@ impl Handler<NetworkClientMessages> for ClientActor {
     type Result = NetworkClientResponses;
 
     fn handle(&mut self, msg: NetworkClientMessages, _: &mut Context<Self>) -> Self::Result {
+        // There is a bug in Actix library. While there are messages in mailbox, Actix
+        // will prioritize processing messages until mailbox is empty. Execution of any other task
+        // scheduled with run_later will be delayed.
+        self.try_handle_block_production();
```
Comment (SkidanovAlex) [inline chain/client/src/client_actor.rs:185]: Shall we also add it to the handlers of `Status` and `GetNetworkInfo` messages below?

Comment (bowenwang1996) [reply chain/client/src/client_actor.rs:185]: agree

**Comment by SkidanovAlex** [comment]
Can you also add similar checks for `chunk_request_retry` and `doomslug_timer`? If they stall, block production will stall as well.

Other `run_later`'s seem to be OK to be delayed (syncing and catchup).

**Comment by pmnoxx** [comment]
@SkidanovAlex Can you take a look at the updated PR?

**Comment by pmnoxx** [comment]
There is one thing I just noticed about Actix. Actix queue can be unbounded or bounded depending on how it's initialized.
By default mailboxes queue, which are used to send messages between actors, are initialized with buffer size of 16, the sender actor will block if receivers mailbox is full.

Hmm, I'm curious now, on how they avoid deadlocks, if there is a pair of actors, which send messages to each other.

File: chain/client/src/client_actor.rs [hunk 2]
```diff
@@ -691,16 +707,80 @@ impl ClientActor {
...
+        self.try_chunk_request_retry(ctx);
+    }
+
+    fn try_handle_block_production(&mut self) {
+        let now = Utc::now();
+        if now > self.block_production_next_attempt {
+            return;
+        }
+        self.block_production_next_attempt = now.checked_sub_signed(
```
[the remark this answers is not in the document]
Comment (pmnoxx) [reply chain/client/src/client_actor.rs]: I fixed that issue, and I also refactored the code to remove code duplication.

**Pull request merged**